dev-docs/RFCs/v7.x/gpu-grid-layer.md

```diff
@@ -0,0 +1,40 @@
+# RFC: GPU Acceleration in GridLayer
+
+* **Authors**: Ravi Akkenapally
+* **Date**: May 2019
+* **Status**: **Draft**
+
+
+## Abstract
+
+GPU accelerated aggregation is currently available in `ScreenGridLayer` and `ContourLayer`. In 6.3 release, `GPUAggregator` is enhanced to support aggregation of multiple weights required for `GridLayer`. This RFC proposes adding GPU accelerated aggregation to `GridLayer`.
+
+
+## Fallback cases
+
+There are several features of `GridLayer` that can't be supported by GPU Aggregation.
+
+* `Lower and higher Percentile` : When these props are set to a non zero value, grid cells are sorted based on their aggregated values and some of the begining and ending cells are ignored (not rendered) based on specified percentile value. On GPU, sorting is not possible.
+
+* `Aggregation operation` : Any aggregation other than `Min`, `Max`, `SUM` or `MEAN` is not supported on GPU.
+
+* `Picking` : When aggregation is performed on CPU, each cell can point to an array of objects that fall into the cell. And when a cell is picked, `GridLayer` can provide this array of objects. But on GPU picking information will be limited to `aggregated value` of the cell and `domain value` (min/max value of all aggregated cells).
```

Comment:
Idea: For better parity, could we offer a "CPU aggregation function" so that the user could optionally calculate these object e.g. when a pick happens?

Reply:
Yes, this should be do able, existing `GridLayer` gets picking data from its state. Data returend on picking includes `colorValue`, `elevationValue` and an `array of original data points`. When on GPU Aggregation we have both `colorValue` and `elevationValue`, for the third we can have a utility method. Some of the existing `GridLayer` code can be re-factored and make this utility method shared between both layers. Will update the RFC.

Reply:
Had a chat with @heshan0131 , it should be fine to provide, current grid cell values on mouse over and provide full list of objects when queried programmatically.